### DISCUSSION TITLE
Break down simp tactics into all tactics used

### DISCUSSION THREAD

**Discussion by realharryhero** [lean-dojo/LeanDojo#199] [Ideas]
A lot of theorems use the tactic "[simp](https://leanprover-community.github.io/extras/simp.html)," which is a tactic that uses many trivial tactics to "simplify" the problem down to an easier version quickly (using identities like a * 1 = a, 1 + 0 = 1, etc). When analyzing the tactics used to solve a particular problem, for example when using `thm.get_traced_tactics()` in LeanDojo, the list of tactics sometimes includes a `simp` tactic. But no info is given to what theorems the `simp` tactic actually uses.

Maybe (maybe) it'd be a good idea to have an option to include what the `simp` tactic uses? Certainly one can figure out a _superset_ of the set of tactics `simp` uses by seeing all tactics `simp` is able to use, but `simp` only uses a subset of those tactics, and that subset probably cannot be checked in LeanDojo itself, as the only way to do so at [the highlighted text here](https://leanprover-community.github.io/extras/simp.html#:~:text=Indeed%2C%20if%20you%20add%20set_option%20trace.Meta.Tactic.simp.rewrite%20true%20above%20the%20example%2C%20then%20a%20squiggly%20blue%20underline%20will%20appear%20under%20simp%20(in%20VS%20Code)%20and%20clicking%20on%20this%20will%20show%20you%20the%20sequence%20of%20rewrites%20that%20simp%20performed%3A) requires something beyond just typing lean tactics to interact with the state. Thank you!

**Comment by yangky11** [comment]
I would favour leaving `simp` as is rather than breaking it into a sequence of `rw`. Tactics in Lean are just meta-programs, and you can easily define new tactics that use existing tactics in the same way `simp` use `rw`. If we choose to break `simp` into `rw`, then we should probably also try to break all tactics into the most atomic ones, which is infeasible.

**Comment by realharryhero** [comment]
I see, fair point. Thank you!
